### PR TITLE
Hopefully resolving some of the Scrutinizer alerts

### PR DESCRIPTION
**Pull request by rawkode**

### CODE REVIEW

**Comment by colinodell** [comment]
Thanks for submitting this!

Scrutinizer failed to run for some reason so I've restarted it.  I'll take a closer look once it completes its analysis.

File: src/DocParser.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ private function incorporateLine(ContextInterface $context)
         $context->setContainer($context->getDocument());
         while ($context->getContainer()->hasChildren()) {
             $lastChild = $context->getContainer()->getLastChild();
-            if (!$lastChild->isOpen()) {
+            if (!is_null($lastChild) && !$lastChild->isOpen()) {
```
Comment (colinodell) [inline src/DocParser.php]: I don't think this check is necessary.  Line 99 checks whether any children exist, so line 100 will always return a child object.

Comment (rawkode) [inline src/DocParser.php]: Ah, you're correct! I'll remove

**Comment by rawkode** [comment]
Scrutinizer seems to be complaining with

```Error while resolving doc type "$regex": Unknown type name "$regex" at position 0.```

**Comment by colinodell** [comment]
I restarted Scrutinizer again (I think it timed out waiting on Travis).  Anyway, these changes look good.  Thanks so much!

**Pull request merged**